.github/workflows/cache-deps.yml

```diff
@@ -0,0 +1,121 @@
+name: Caching of dependencies
+
+#  2021-11-30: NOTE: This workflow currently a trimmed copy of a main `test.yml` workflow. Workflows need further deduplication: https://docs.github.com/en/actions/learn-github-actions/reusing-workflows#overview
```

Comment:
I am a little bit worried too by the duplication between this and the test workflow (to the already existing duplication with the build workflow). But well we can tackle it in next pr's
Will take a look to the link you provided to see how can we remove it
have you planned working on that?